server/channels/app/websocket_broadcast_hooks.go

```diff
@@ -0,0 +1,59 @@
+package app
+
+import (
+	"github.com/mattermost/mattermost/server/public/model"
+	pUtils "github.com/mattermost/mattermost/server/public/utils"
+	"github.com/mattermost/mattermost/server/v8/channels/app/platform"
+)
+
+const AddMentionsAndFollowers = "add_mentions_and_followers"
+
+func (s *Server) RegisterBroadcastHooks() {
+	broadcastHooks := map[string]*platform.BroadcastHook{
+		AddMentionsAndFollowers: makeAddMentionsAndFollowersHook(),
+	}
+
+	s.platform.HubsUseBroadcastHooks(broadcastHooks)
+}
+
+func makeAddMentionsAndFollowersHook() *platform.BroadcastHook {
+	return &platform.BroadcastHook{
+		HasChanges: func(msg *model.WebSocketEvent, webConn *platform.WebConn, args map[string]any) bool {
+			if msg.EventType() != model.WebsocketEventPosted {
+				return false
+			}
+
+			mentions, _ := args["mentions"].(model.StringArray)
```

Comment:
I was trying to make sure that this code was type safe, but because I couldn't figure out how to make the type of `args` narrower than `map[string]any`, even with generics, I went with just adding the second return value to suppress panics if the argument is missing

Reply:
I can change it to be more idiomatic then. I would like if I could make this type check happen as a compile time check to avoid needing any of this at all, but I'm still not sure how to accomplish that even with generics.

If I do that, should I change this method to return an error if that happens, or should I just log an error and move on when it occurs?